std/assembly/string.ts

```diff
@@ -127,6 +127,83 @@ export class String {
     );
   }
 
+  @operator("!=")
+  private static __ne(left: String, right: String): bool {
+    return !this.__eq(left, right);
+  }
+
+  @operator(">")
+  private static __gt(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    var length = <usize>min<i32>(left.length, right.length);
+    if (length == 0) {
+      return left.charCodeAt(0) > right.charCodeAt(0);
+    }
+
+    return compare_memory(
+      changetype<usize>(left)  + HEADER_SIZE,
+      changetype<usize>(right) + HEADER_SIZE,
+      length << 1
+    ) > 0;
+  }
+
+  @operator(">=")
+  private static __gte(left: String, right: String): bool {
+    if (!changetype<usize>(left) || !changetype<usize>(right)) {
+      return false;
+    }
+
+    var length = <usize>min<i32>(left.length, right.length);
+    if (length == 0) {
+      return left.charCodeAt(0) >= right.charCodeAt(0);
```

Comment:
doesn't this result in an out of bounds read on the zero-length one?

Reply:
No, because string with zero length actually is "\0"

Reply:
Is it? Should be i32-length + u16 code units in memory, not zero-terminated.